src/main/java/stream/flarebot/flarebot/util/general/GeneralUtils.java

```diff
@@ -191,194 +401,38 @@ public static AudioItem resolveItem(Player player, String input) throws IllegalA
-                    Member temMember = Getters.getGuildById(guildId).getMemberById(l);
-                    if (temMember != null) {
-                        tmp = temMember.getUser();
-                    }
-                }
-                if (tmp != null) {
-                    return tmp;
-                } else if (forceGet) {
-                    return Getters.retrieveUserById(l);
-                }
-            } catch (NumberFormatException | NullPointerException ignored) {
-            }
-        }
-        return null;
-    }
-
-    public static Role getRole(String s, String guildId) {
-        return getRole(s, guildId, null);
-    }
-
-    public static Role getRole(String s, String guildId, TextChannel channel) {
-        Guild guild = Getters.getGuildById(guildId);
-        Role role = guild.getRoles().stream()
-                .filter(r -> r.getName().equalsIgnoreCase(s))
-                .findFirst().orElse(null);
-        if (role != null) return role;
-        try {
-            role = guild.getRoleById(Long.parseLong(s.replaceAll("[^0-9]", "")));
-            if (role != null) return role;
-        } catch (NumberFormatException | NullPointerException ignored) {
-        }
-        if (channel != null) {
-            if (guild.getRolesByName(s, true).isEmpty()) {
-                String closest = null;
-                int distance = LEVENSHTEIN_DISTANCE;
-                for (Role role1 : guild.getRoles().stream().filter(role1 -> FlareBotManager.instance().getGuild(guildId).getSelfAssignRoles()
-                        .contains(role1.getId())).collect(Collectors.toList())) {
-                    int currentDistance = StringUtils.getLevenshteinDistance(role1.getName(), s);
-                    if (currentDistance < distance) {
-                        distance = currentDistance;
-                        closest = role1.getName();
-                    }
-                }
-                MessageUtils.sendErrorMessage("That role does not exist! "
-                        + (closest != null ? "Maybe you mean `" + closest + "`" : ""), channel);
-                return null;
-            } else {
-                return guild.getRolesByName(s, true).get(0);
-            }
-        }
-        return null;
-    }
-
-    public static TextChannel getChannel(String arg) {
-        return getChannel(arg, null);
-    }
-
-    public static TextChannel getChannel(String channelArg, GuildWrapper wrapper) {
-        try {
-            long channelId = Long.parseLong(channelArg.replaceAll("[^0-9]", ""));
-            return wrapper != null ? wrapper.getGuild().getTextChannelById(channelId) : Getters.getChannelById(channelId);
-        } catch (NumberFormatException e) {
-            if (wrapper != null) {
-                List<TextChannel> tcs = wrapper.getGuild().getTextChannelsByName(channelArg, true);
-                if (!tcs.isEmpty()) {
-                    return tcs.get(0);
+    /**
+     * Checks if a perm is valid.
```

Comment:
How does it? What perms can be valid? What happens with *? Why is that not included?